tests/monotouch-test/ModelIO/MDLObject.cs

```diff
@@ -63,6 +63,19 @@ public void GetBoundingBox ()
 				var bb = obj.GetBoundingBox (0);
 			}
 		}
+
+		[Test]
+		public void IndexerTest ()
+		{
+			using (var obj = new MDLObject ()) {
+				var key = new Protocol ("MDLComponent");
```

Comment:
I'm not sure we want to expose `Protocol` like this in the API. At best it's not very type safe, I'd better use

> var key = new Protocol (typeof (MDLComponent));

My first thought was to _abuse_ `[BindAs]` but a conversion from `Protocol` to `System.Type`, with additional checks to ensure it's a protocol, can be costly if done repetitively...  

Maybe we can just provide a `[Wrapper]` that calls a conversion method ? and also expose that as a new `Protocol.ctor(System.Type)` ? 

@rolfbjarne @dalexsoto thoughts ?